src/base/memory_monitor_sr.cpp

```diff
@@ -20,18 +20,58 @@
  * memory_monitor_sr.cpp - Implementation of memory monitor module
  */
 
+#include <malloc.h>
+#include <cassert>
 #include <cstring>
 #include <algorithm>
-#include <cassert>
 
 #include "memory_monitor_sr.hpp"
 
+#ifndef HAVE_USR_INCLUDE_MALLOC_H
+#define HAVE_USR_INCLUDE_MALLOC_H
+#endif
+
+typedef struct mmon_metainfo MMON_METAINFO;
+struct mmon_metainfo
+{
+  uint64_t allocated_size;
+  int tag_id;
+  int magic_number;
+};
+
 namespace cubmem
 {
+  static const std::string magic_string ("MMON");
+  memory_monitor *mmon_Gl = nullptr;
+
   memory_monitor::memory_monitor (const char *server_name)
-    : m_server_name {server_name} {}
+    : m_server_name {server_name}
+  {
+    memcpy (&m_magic_number, magic_string.c_str (), sizeof (int));
+  }
+
+  size_t memory_monitor::get_allocated_size (const char *ptr)
+  {
+    size_t allocated_size = malloc_usable_size ((void *)ptr);
+    const char *meta_ptr = ptr + allocated_size - MMON_ALLOC_META_SIZE;
+
+    if (allocated_size <= MMON_ALLOC_META_SIZE)
+      {
+	return allocated_size;
+      }
 
-  static std::string memory_monitor::make_tag_name (const char *file, const int line)
+    assert (meta_ptr > ptr);
```

Comment:
This can be removed with the `if-return` above.
```suggestion
```

Reply:
Done.